app/assets/javascripts/discourse/app/components/d-editor.js

```diff
@@ -904,6 +905,21 @@ export default Component.extend({
     });
   },
 
+  @observes("value")
+  normalizeEmoji() {
+    let inputString = this.value;
+    inputString = inputString.replace(
+      /(?:[\u2700-\u27bf]|(?:\ud83c[\udde6-\uddff]){2}|[\ud800-\udbff][\udc00-\udfff])[\ufe0e\ufe0f]?(?:[\u0300-\u036f\ufe20-\ufe23\u20d0-\u20f0]|\ud83c[\udffb-\udfff])?(?:\u200d(?:[^\ud800-\udfff]|(?:\ud83c[\udde6-\uddff]){2}|[\ud800-\udbff][\udc00-\udfff])[\ufe0e\ufe0f]?(?:[\u0300-\u036f\ufe20-\ufe23\u20d0-\u20f0]|\ud83c[\udffb-\udfff])?)*/gi,
```

Comment:
Can you extract it into a constant? And perhaps split it into multiple lines/const if possible? Similar how [lodash does it](https://github.com/lodash/lodash/blob/e2555a43adfd8e38864cdaf6e0a1f828d38ccc0a/.internal/unicodeWords.js#L46). It will make it easier to see at a glance what ranges are matched, and which one should be expanded if any new emoji are added. 🙂